### PR TITLE
Add growfs service to chatbot/bootc Containerfiles

### PR DESCRIPTION
**Pull request by sallyom**
@dwalsh  @cgwalters testing this now, added colin as author bc i copied his files from bootc-org/examples 

Tested with an `ami` with (and also with `podman-bootc`):
```
$ cd recipes/natural_language_processing/chatbot
$ make BOOTC_IMAGE=quay.io/sallyom/centos-bootc:chatbot ARCH=x86_64 CONTAINERFILE=Containerfile.nocache bootc
$ aws s3 cp ~/Desktop/bootc-disk-images/image/disk.raw s3://BUCKET_NAME
$ aws ec2 import-snapshot  --description "centos-bootc-chatbot" --disk-container file://ami-container.json
$ aws ec2 describe-import-snapshot-tasks --filters Name=task-state,Values=active

# create image (ami) from snapshot in console

$ cat ami-container.json 
{
   "Description": "centos-bootc-chatbot",
   "Format": "raw",
   "Url": "s3://BUCKET_NAME/disk.raw"
}
```

### CODE REVIEW

**Comment by sallyom** [comment]
oops: 
```
[root@ip-172-31-69-31 ~]# systemctl status bootc-generic-growpart.service 
× bootc-generic-growpart.service - Fallback root filesystem growing
     Loaded: loaded (/usr/lib/systemd/system/bootc-generic-growpart.service; static)
     Active: failed (Result: exit-code) since Tue 2024-04-30 21:34:08 UTC; 3min 1s ago
   Duration: 5ms
    Process: 654 ExecStart=/usr/libexec/bootc-generic-growpart (code=exited, status=203/EXEC)
   Main PID: 654 (code=exited, status=203/EXEC)
        CPU: 5ms

Apr 30 21:34:08 localhost systemd[1]: Started Fallback root filesystem growing.
Apr 30 21:34:08 localhost systemd[654]: bootc-generic-growpart.service: Failed to locate executable /usr/libexec/bootc-generic-growpart: Permission denied
Apr 30 21:34:08 localhost systemd[654]: bootc-generic-growpart.service: Failed at step EXEC spawning /usr/libexec/bootc-generic-growpart: Permission denied
Apr 30 21:34:08 localhost systemd[1]: bootc-generic-growpart.service: Main process exited, code=exited, status=203/EXEC
Apr 30 21:34:08 localhost systemd[1]: bootc-generic-growpart.service: Failed with result 'exit-code'.
```

**Comment by sallyom** [comment]
@cgwalters I had to update the bootc-generic-growpart script to get things working - I added some echo statements for a nice `systemctl status`(added for debugging but i like seeing the values):
```$ ssh -i ~/.ssh/id_rsa root@34.238.208.111
Last login: Wed May  1 03:32:22 2024 from 173.76.173.190
[root@ip-172-31-67-89 ~]# systemctl status bootc-generic-growpart
○ bootc-generic-growpart.service - Fallback root filesystem growing
     Loaded: loaded (/usr/lib/systemd/system/bootc-generic-growpart.service; static)
     Active: inactive (dead) since Wed 2024-05-01 03:32:14 UTC; 1min 12s ago
   Duration: 2.812s
    Process: 654 ExecStart=/usr/libexec/bootc-generic-growpart (code=exited, status=0/SUCCESS)
   Main PID: 654 (code=exited, status=0/SUCCESS)
        CPU: 129ms

May 01 03:32:11 localhost bootc-generic-growpart[654]: Backing device: /dev/nvme1n1p4
May 01 03:32:11 localhost bootc-generic-growpart[654]: syspath: /sys/class/block/nvme1n1p4
May 01 03:32:11 localhost bootc-generic-growpart[654]: partition: 4
May 01 03:32:11 localhost bootc-generic-growpart[654]: parentpath: /sys/devices/pci0000:00/0000:00:04.0/nvme/nvme1/nvme1n1
May 01 03:32:11 localhost bootc-generic-growpart[654]: device: /dev/nvme1n1
May 01 03:32:11 localhost bootc-generic-growpart[654]: devmajmin: 259:0
May 01 03:32:11 localhost bootc-generic-growpart[654]: parent: /sys/dev/block/259:0
May 01 03:32:12 localhost bootc-generic-growpart[663]: CHANGED: partition=4 start=3127296 old: size=17844191 end=20971486 new: size=185616351 end=188743646
May 01 03:32:14 ip-172-31-67-89 systemd-growfs[726]: Successfully resized "/sysroot" to 88.5G bytes (3584 bytes lost due to blocksize).
May 01 03:32:14 ip-172-31-67-89 systemd[1]: bootc-generic-growpart.service: Deactivated successfully.
```

**Comment by rhatdan** [comment]
Would it make sense to just use cloud-init for this use case?

File: recipes/natural_language_processing/summarizer/app/summarizer.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 from langchain.prompts import PromptTemplate
 from langchain_community.callbacks import StreamlitCallbackHandler
 from langchain_community.document_loaders import PyMuPDFLoader
+from langchain_text_splitters import RecursiveCharacterTextSplitter
```
Comment (cgwalters) [inline recipes/natural_language_processing/summarizer/app/summarizer.py]: This change looks unrelated?

**Comment by cgwalters** [comment]
> Would it make sense to just use cloud-init for this use case?

Unfortunately at the current time that won't help

**Review by cgwalters — APPROVED** [review]
We're almost certainly going to add this functionality to the base image soon btw.

File: recipes/natural_language_processing/chatbot/bootc/Containerfile [hunk 2]
```diff
@@ -19,6 +19,11 @@ ARG APP_IMAGE=quay.io/ai-lab/${RECIPE}:latest
 ARG SERVER_IMAGE=quay.io/ai-lab/llamacpp_python:latest
 ARG TARGETARCH
 
+# Include growfs service
+COPY build/usr/lib /usr/lib
+COPY build/usr/libexec /usr/libexec
+RUN chmod +x /usr/libexec/bootc-generic-growpart
```
Comment (cgwalters) [inline recipes/natural_language_processing/chatbot/bootc/Containerfile]: Very minor nit, it is better to just make it executable in git.

File: recipes/natural_language_processing/chatbot/bootc/Containerfile.nocache [hunk 3]
```diff
@@ -15,6 +15,11 @@ RUN set -eu; mkdir -p /usr/ssh && \
 
 ARG RECIPE=chatbot
 
+# Include growfs service
+COPY build/usr/lib /usr/lib
+COPY build/usr/libexec /usr/libexec
+RUN chmod +x /usr/libexec/bootc-generic-growpart
```
Comment (rhatdan) [inline recipes/natural_language_processing/chatbot/bootc/Containerfile.nocache]: Ditto to change in git.

File: recipes/natural_language_processing/chatbot/bootc/Containerfile (hunk 2 above)
Comment (sallyom) [reply recipes/natural_language_processing/chatbot/bootc/Containerfile]: updated

File: recipes/natural_language_processing/chatbot/bootc/Containerfile.nocache (hunk 3 above)
Comment (sallyom) [reply recipes/natural_language_processing/chatbot/bootc/Containerfile.nocache]: updated

File: recipes/natural_language_processing/summarizer/app/summarizer.py (hunk 1 above)
Comment (sallyom) [reply recipes/natural_language_processing/summarizer/app/summarizer.py]: 👍 I removed the extra commit

**Comment by sallyom** [comment]
@rhatdan @cgwalters I used [this PR](https://gitlab.com/fedora/bootc/base-images/-/merge_requests/9) to update the script, ptal
Here's the output (with added idempotency):
```
# systemctl status bootc-generic-growpart
○ bootc-generic-growpart.service - Fallback root filesystem growing
     Loaded: loaded (/usr/lib/systemd/system/bootc-generic-growpart.service; static)
     Active: inactive (dead) since Wed 2024-05-01 13:00:48 UTC; 6min ago
   Duration: 137ms
    Process: 659 ExecStart=/usr/libexec/bootc-generic-growpart (code=exited, status=0/SUCCESS)
   Main PID: 659 (code=exited, status=0/SUCCESS)
        CPU: 79ms

May 01 13:00:48 localhost systemd[1]: Started Fallback root filesystem growing.
May 01 13:00:48 localhost bootc-generic-growpart[659]: Backing device: /dev/nvme0n1p4
May 01 13:00:48 localhost bootc-generic-growpart[709]: NOCHANGE: partition 4 is size 185616351. it cannot be grown
May 01 13:00:48 localhost systemd[1]: bootc-generic-growpart.service: Deactivated successfully.
```

File: recipes/natural_language_processing/chatbot/bootc/Containerfile (hunk 2 above)
Comment (sallyom) [reply recipes/natural_language_processing/chatbot/bootc/Containerfile]: I found I had to add `--chmod=0755` to the `COPY` directive still.

**Review by cgwalters — APPROVED** [review]

**Comment by sallyom** [comment]
@rhatdan ready to merge? I've pulled in the latest from [gitlab.com/fedora/base-images#9](https://gitlab.com/fedora/bootc/base-images/-/merge_requests/9)


I'll make an issue to add growfs to audio_to_text, summarizer, codegen, and rag bootc/Containerfiles if this merges

**Comment by rhatdan** [comment]
LGTM
